File: algorithm/src/main/java/baekjoon/p03049/Main.java
package baekjoon.p03049;

import java.util.Scanner;

public class Main {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.println(solve(sc.nextInt()));
    }

    private static int solve(int n) {
        return (n * (n - 1) * (n - 2) * (n - 3)) / (4 * 3 * 2);
    }

}

File: algorithm/src/main/java/baekjoon/p03049/Main.kt
package baekjoon.p03049

fun main(args: Array<String>) {
    println(solve(readLine()!!.toInt()))
}

private fun solve(n: Int): Int {
    return mul(n downTo n - 3) / fac(4)
}

private fun mul(range: IntProgression): Int {
    var ret = 1

    for (n in range) {
        ret *= n
    }

    return ret
}

private fun fac(n: Int): Int {
    return when (n <= 1) {
        true -> 1
        false -> n * fac(n - 1)
    }
}